components/popover_list_members/popover_list_members.tsx

```diff
@@ -14,34 +13,46 @@ import {localizeMessage} from 'utils/utils.jsx';
 import {trackEvent} from 'actions/telemetry_actions.jsx';
 
 import {AddMembersToChanneltreatments} from 'mattermost-redux/constants/config';
+import {Channel} from 'mattermost-redux/types/channels';
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {ModalData} from 'types/actions';
 
 import ChannelMembersModal from 'components/channel_members_modal';
 import ChannelInviteModal from 'components/channel_invite_modal';
 import OverlayTrigger from 'components/overlay_trigger';
 import Tooltip from 'components/tooltip';
 import Popover from 'components/widgets/popover';
-
 import PopoverListMembersItem from 'components/popover_list_members/popover_list_members_item';
-
-export default class PopoverListMembers extends React.PureComponent {
-    static propTypes = {
-        channel: PropTypes.object.isRequired,
-        statuses: PropTypes.object.isRequired,
-        users: PropTypes.array.isRequired,
-        memberCount: PropTypes.number,
-        currentUserId: PropTypes.string.isRequired,
-        teamUrl: PropTypes.string,
-        manageMembers: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            openModal: PropTypes.func.isRequired,
-            loadProfilesAndStatusesInChannel: PropTypes.func.isRequired,
-            openDirectChannelToUserId: PropTypes.func.isRequired,
-        }).isRequired,
-        sortedUsers: PropTypes.array,
-        addMembersABTest: PropTypes.string,
+import {RelationOneToOne} from 'mattermost-redux/types/utilities';
```

Comment:
also not related, but IMO the `RelationOneToOne` is not very helpful. all it basically does is checking for the first generic type to have the property `id` and assign a new generic property (which always will be `string`) to the second generic property as seen [here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wFgAoCmATzCUSQBsUZgIA7AeXaQBUJuSADwBROEgAeMJOwAmAZzgBvYLIBcceTCjB2AcwC+AGji8AfHAC8yinDtwA2hLi64Ih-lX4Aut428AbgoDIMpyGjo4AFV5JCgrZVUNLR19ALh2FBAkZO1dPQCDCgo0Di04aS0ARg1kZlYOQX5BIRi4k3YAVxAAIziLaxV1OCqAJhCSsvhKmFHaphY2Lh5mnlbYqA7uvqgBxOGxjqycggApCAALdnwQoA)

Reply:
Thanks for additional info